link-area-delegation-explainer.md

```diff
@@ -0,0 +1,141 @@
+Today, achieving this is done using two alternatives, each with its own tradeoffs:
+1. Stacking: put the link *behind* the buttons, so that it catches clicks that are not handled by the buttons.
+2. Capturing: have the card capture clicks, and delegate them to the main link only if the original target is not one of the buttons.\
+
+These are imperfect and brittle ways to achieve this kind of experience, however this became a common enough pattern in the web, so perhaps it is time it became a first class citizen.
+
+## Don't panic (a11y concerns)
+The ARIA guidelines discussed earlier are still valid, so this needs to be treaded carefully. However, it doesn't seem like the websites that introduce this pattern break ARIA in any way.
+That's because this pattern doesn't modify the interactive structure of the page (AKA semantics, AKA the accessibility tree), it only gives cards enhanced behavior when they are clicked,
+allowing their clicks to be handled as if its target was the main link.
+
+One accessibility issue that can be caused by using this (and any of the existing methods) is creating clickable targets that are too close together (see [these guidelines](https://www.w3.org/WAI/perspective-videos/controls/)).
+This is indeed a concern, but it's debatable whether enabling this feature in HTML makes a difference for this concern vs. the current approaches.
+The contrary might be true, where giving this information to the UA might give it a chance to intervene when the clickable nested target is too small.
+
+## Link Area Delegation
+The proposed mental model of this problem is "the card is delegating its interactive area to one of its link descendants".
+This can mean two things in terms of UI:
+1. The default click action for the card is the default click action of the link descendant
+2. The card's context menu behavior is augmented with the link descendant's context menu behavior (e.g. "Open in a new tab")
+3. Potentially, the UA can expand the hit-testing area for some of the descendants, to prioritize clicking them vs. clicking on the card.
+4. This feature has no effect on keyboard navigation.
+6. At first, this feature is not exposed to the accessibility tree, and should have no effect on assistive technologies. We can research in the future whether this card/link relationship can be expressed in these technologies in a useful way.
+
+### Link delegation attributes
+Expressing link area delegation in HTML can look something like this:
+```html
+<section class="card" linkarea>
+  <a href="/post?id=123" defaultlink>Post Title</a>
+  <img>
+  <button>Join</button>
+  <button>Share</button>
+</section>
+```
+
+In the above example, the `section` element has a `linkarea` attribute, which would delegate its click to the first descendant with a `defaultlink` attribute, in this case the first `a` element.
+
+We could have different similar alternatives to this with different ergonomic tradeoffs, e.g. using ID references:
+```html
```

Comment:
I think we should make this the primary proposal.

The `linkarea`/`defaultlink` approach is convenient but it creates the potential for additional edge-cases. For example, if two links within the `linkarea` claim to be the `defaultlink`.

Reply:
I think both of those approaches can co-exist. Anyway I'm rewording to be less opinionated about which is the primary proposal.